app/models/course/registration.rb

```diff
@@ -0,0 +1,29 @@
+class Course::Registration < ActiveRecord::Base
+  stampable
+
+  scope :staff, -> { where(role: roles.values_at(:teaching_assistant, :manager, :owner)) }
```

Comment:
...or maybe it's just simpler to declare a fixed hash containing the role categories, maybe `CourseUser.role_categories` so we use `CourseUser.role_categories[:staff]`, if that reduces the likelihood of messing ActiveRecord up.